submissions/Temu4/js_dom/script.js

```diff
@@ -0,0 +1,75 @@
+//Toggle sidemenu
+const sideMenu = document.getElementById('sideMenu');
```

Comment:
Avoid getting element by id. Better way use class.

Reply:
Please, explain?

Reply:
@Temu4  We learn to write code for large and good structured projects. `querySelector` more useful when you want to use more complex selectors in future. Also i see you try to use BEM for html structure. This approach prefer use class name selector. Also remember that IDs are unique identifiers, and for that reason should have more specific names. You can use them only when they are actually necessary and when a class wouldn't be a better way to identify them. Some examples from your code:
```javascript
<nav class="container__side-menu side-menu" id="sideMenu">
``` 
In this case you have perfect element with `side-menu` class. `id` attribute with repeated `side menu` is not necessary. `id` doesn't explain unique element.
```javascript
<button class="toggle-button" id="toggleButton"> 
```
Just use `querySelector` for access to element. `id` doesn't unique also and is not necessary.

Reply:
@o-msh, thanks for you answer. Let I explain, why I've used such class name:
```html
<nav class="container__side-menu side-menu">
```
```css
.side-menu {
  width: 250px;
  height: auto;
  background-color: transparent;
  color: #efefef;
}
```
is responsible for element itself, but
```css
.container__side-menu {
  margin-top: 25px;
  position: absolute;
  -webkit-transform: translate(-300px, 0);
  transform: translate(-300px, 0);
  transition: transform 0.3s ease;
}
```
describes its behavior(position) inside the parent element ("container"). 
I tried separate these and don't mix it, for future performance (if I'll need add the same element at another position in the page).
Also, thank you for mentioning the about ```id```, honestly speaking, I don’t like this approach either, but I didn’t know if it’s right to use ```class``` to select an element (because they can be repeated).

Reply:
@Temu4 Better will be use selection by `class`. So `querySelector` will be preferred. Even if you want to add some repeated elements in future, you must check yourself for right BEM with well structured and nested selectors. In our case we have a simple project with sufficient functional.

Reply:
- [x] issue with id fixed in [this](https://github.com/kottans/frontend-2019-p2p/pull/42/commits/5807b3f2b4609bd55aa7bb6ad18d359c187f6132) commit;
- [x] issue with classes fixed in [this](https://github.com/kottans/frontend-2019-p2p/pull/42/commits/503cae9b959bde4a6a1700542f3d3d61b1d87a45) commit.
Anything else?